plugins/calibrecompanion.koplugin/main.lua

```diff
@@ -49,6 +49,44 @@ local function delete(file)
     end
 end
 
+-- disk usage
+local function diskUsage(dir)
+    -- fallback to 2GB total, 1GB used, 1GB free
+    local function diskUsageFallback()
+        return {
+            total = 2 * 1024 * 1024 * 1024,
+            used = 1024 * 1024 * 1024,
+            available = 1024 * 1024 * 1024,
+        }
+    end
+    -- we only work in valid directories
+    if not dir or lfs.attributes(dir, "mode") ~= "directory" then
+        return diskUsageFallback()
+    end
+    -- fill a table with total, available and used kb of the mountpoint that holds the directory,
+    -- this relies on common shell utilities: cut, df, grep & sed.
+    local std_out =  io.popen(
+        "df "..dir.." 2>/dev/null | sed -r 's/ +/ /g' "..
+        " | grep /dev | cut -d ' ' -f 2,3,4"
```

Comment:
Pinging @NiLuJe about this one. Does it work on kobos, kindles?. It works on cervantes and should work as well in the Remarkable. It certainly works on desktop.

Reply:
See the system stats plugin.

Reply:
Or was it a function in util (am on phone)

Reply:
system stats isues a general df command and greps the directory. Here we rely on df to figure out the mountpoint of a given dir. It should work fine even on busybox df but I'm not sure.

Reply:
```
# df /mnt/onboard 2>/dev/null | sed -r 's/ +/ /g' | grep /dev | cut -d ' ' -f 2,3,4
3316908 2945656 371252
```

Reply:
If you want a battle-tested (MRPI) variant with fewer forks:

```
df -k /mnt/onboard | awk '$3 ~ /[0-9]+/ { print $2,$3,$4 }'
```